### PR TITLE
Add PULUMI_SKIP_CONFIRMATIONS env var

### PR DESCRIPTION
**Pull request by chrsmith**
Add a new environment variable `PULUMI_SKIP_CONFIRMATIONS` which will act as if you had added the `--yes` flag to applicable commands. This provides a global switch for disabling interactive prompts.

I can't say I'm especially happy with the design here, but I'm not sure what would look better since we are already treating `--non-interactive` to mean "don't prompt for input, but still require `--yes` to confirm intent".

What do you think? Does this seem like the right way to fix https://github.com/pulumi/pulumi/issues/4459 and related issues? Or is there a better or more appropriate way to disable prompting?

```
// DisableInteractive may be set to true in order to disable prompts. This is useful when running in a non-attended
// scenario, such as in continuous integration, or when using the Pulumi CLI/SDK in a programmatic way.
var DisableInteractive bool

// Interactive returns true if we should be running in interactive mode. That is, we have an interactive terminal
// session, interactivity hasn't been explicitly disabled, and we're not running in a known CI system.
func Interactive() bool {
	return !DisableInteractive && InteractiveTerminal() && !ciutil.IsCI()
}
```

### CODE REVIEW

**Comment by lukehoban** [comment]
> Reading through the PR and issue, I think this change is inline with the intent we had (which is to remove implicitly assuming --yes). Today, we only have one way to explicitly say --yes which is via the args. Having an env var seems like a nice convenience for power users. The only negative here would be someone who sets the env var in a non-CI environment, forgets they did so, and gets unintended behavior. But this seems like the right balance to me.

> I'd love to get the opinions from the folks who were on the original PR to see if I'm missing anything in the reasoning above /cc @lukehoban @justinvp @pgavlin

Yes - that all seems in line with the original intent.  The reason for the change was just that the default here is dangerous for manual usage at a terminal (things like piping to a file inadvertently auto-accepting a prompt).  For CI use cases I think it's reasonable to opt-in to this auto-accept default, as they have more well-defined use cases encoded in the actual CI workflows.

File: pkg/cmd/pulumi/util.go [hunk 1]
```diff
@@ -65,6 +65,16 @@ func useLegacyDiff() bool {
...
 }
 
+// skipConfirmations returns whether or not confirmation prompts should
+// be skipped. This should be used by pass any requirement that a --yes
+// parameter has been set for non-interactive scenarios.
+//
+// This should NOT be used to bypass protections for destructive
+// operations, such as those that will fail without a --force parameter.
+func skipConfirmations() bool {
+	return cmdutil.IsTruthy(os.Getenv("PULUMI_SKIP_CONFIRMATIONS"))
```
Comment (justinvp) [inline pkg/cmd/pulumi/util.go:75]: Reminder to open issue/PR to update: https://github.com/pulumi/docs/blob/master/content/docs/reference/cli/environment-variables.md

**Comment by stack72** [comment]
Going to merge this so we can proceed for the 2.2.0 release

File: pkg/cmd/pulumi/util.go (hunk 1 above)
Comment (stack72) [reply pkg/cmd/pulumi/util.go:76]: https://github.com/pulumi/docs/pull/3313

**Pull request merged**